wcomponents-theme/src/test/spec/wc.ui.redirect.test.mjs

```diff
@@ -0,0 +1,81 @@
+import redirect from "wc/ui/redirect.mjs";
+
+describe("wc/ui/redirect", () => {
+	// window = Object.create(Window);
+
+	const delay = 50;  // milliseconds to wait for events and stuff to be actioned
+	const origWindow = window.parent;
+	const replaceSpy = jasmine.createSpy();
+
+	beforeEach(() => {
+		window.parent = {
+			location : {
+				storedUrl : "",
+				set href(url) {
+					this.storedUrl = url;
+				},
+				replace : function (url) {
+					this.href = url;
+				}
+			}
+		};
+	});
+
+	it("does nothing with an empty url", () => {
+		spyOnProperty(window.parent.location, "href", "set")
+			.and.throwError("href should not be getting touched for empty url");
+		spyOn(redirect, "register").and.callThrough();
+		redirect.register("");
+		expect(redirect.register).toHaveBeenCalledTimes(1);
+	});
+
+	it("redirects to the given non-launch url", () => {
+		spyOn(redirect, "register").and.callThrough();
+		return testRegister("https://github.com/BorderTech/wcomponents", false);
+	});
+
+	it("does not redirect to new url given a launch url", () => {
+		spyOn(redirect, "register").and.callThrough();
+		return testRegister("mailto:test@email.com", true);
+	});
+
+	it("recognises attachment links as launch links", () => {
+		spyOn(redirect, "isLaunchUrl").and.callThrough();
```

Comment:
I think you may be overusing the spy here (and elsewhere in this suite)? What value does it add?
For example, this test explicitly calls `redirect.isLaunchUrl` and then uses a spy to verify that this call happened.
Could this not be simplified to something like this?

```js
	it("recognises attachment links as launch links", () => {
		expect(redirect.isLaunchUrl("mailto:test@email.com")).toBeTrue();
	});
```